core-tests/src/test/java/org/mule/runtime/core/internal/construct/DefaultFlowTestCase.java

```diff
@@ -524,4 +564,18 @@ public CoreEvent process(CoreEvent event) throws MuleException {
           });
     }
   }
+
+  public static class BlockMessageProcessor extends AbstractComponent implements Processor {
+
+    @Override
+    public CoreEvent process(CoreEvent event) throws MuleException {
+      return sleepFor(event, muleContext.getConfiguration().getShutdownTimeout() * 2);
```

Comment:
Isn't it possible to use a Latch?

Reply:
Discussed offline, but short answer is that a latch has no use for this test case (we just couldn't know when such latch should be released)